controllers/offer.js

```diff
@@ -15,13 +15,19 @@ const getOfferById = async (req, res) => {
 }
 
 const createOffer = async (req, res) => {
-  const { id: userId } = req.params
+  const { id: authorId } = req.params
   const { role: authorRole } = req.user
+  const { price, proficiencyLevel, description, languages, subjectId, categoryId } = req.body
```

Comment:
you have way too many fields to repeat, prolly its better to pass here just

const offer = req.body

const newOffer = await offerService.createOffer({
    authorRole,
    authorId,
    offer
  })

Reply:
refactored